crates/evm/src/execute.rs

```diff
@@ -209,18 +213,23 @@ pub trait BlockExecutionStrategy<DB> {
     fn state_mut(&mut self) -> &mut State<DB>;
 
     /// Sets a hook to be called after each state change during execution.
-    fn with_state_hook(&mut self, hook: Option<Box<dyn OnStateHook>>);
+    fn with_state_hook(&mut self, _hook: Option<Box<dyn OnStateHook>>) {}
```

Comment:
this a bit weird that we allow this if the hook can be ignored.
but I think for now this is okay.